src/client.rs

```diff
@@ -142,8 +142,56 @@ pub struct CachedFeature {
     // on submission will be the next logical progression.
     enabled: AtomicU64,
     disabled: AtomicU64,
+    disabled_variant_count: AtomicU64,
     // Variants for use with get_variant
-    variants: Vec<api::Variant>,
+    variants: Vec<CachedVariant>,
+}
+
+impl CachedFeature {
+    #[allow(dead_code)]
+    fn variant_metrics(&self) -> HashMap<String, u64> {
+        self.variants
+            .iter()
+            .map(|variant| (variant.name.clone(), variant.count.load(Ordering::Relaxed)))
+            .chain([(
+                "disabled".into(),
+                self.disabled_variant_count.load(Ordering::Relaxed),
+            )])
+            .collect()
+    }
+}
+
+#[derive(Default)]
+pub struct CachedVariant {
+    count: AtomicU64,
```

Comment:
I suggest you don't put the name/weight/paylog/overrides in here. Instead nest an api::Variant.

```
pub struct CachedVariant {
    count: AtomicU64,
    variant: api::Variant
}
```

Reply:
Sure thing! I figured that `CachedFeature` re-implements all of its fields, so figured I'd do the same for variant, but there may be other considerations.